src/test/java/com/google/api/codegen/util/ProtoParserTest.java

```diff
@@ -0,0 +1,156 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.Retry;
+import com.google.api.codegen.CodegenTestUtil;
+import com.google.api.codegen.protoannotations.GapicCodeGeneratorAnnotationsTest;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.MessageType;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.ProtoFile;
+import com.google.api.tools.framework.model.testing.TestDataLocator;
+import com.google.longrunning.OperationTypes;
+import com.google.rpc.Code;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class ProtoParserTest {
```

Comment:
Why does `ProtoParser` class uses a mix of static/non-static methods in a way it does (I can't understand a pattern)? Currently `ProtoParser` looks like a mix of a singleton, stateless helper class (class with public constructor but not data/fields/state in it) and a utility class with static methods only:
- it has `getProtoParser()` (i.e. `getInstance()`) - an attribute of a singleton (though constructor is public);
- it has non-static methods with some functionality in them (not just getters/setters), a public constructor and no state/fields - an attribute of a stateless helper class;
- it has static methods with some functionality, not related to instantiating the class itself (or doing something specific, which forces the method to be static) - an attribute a utilities class with static-only methods;

 Specifically, why `getResourceEntityName(Field)` is static and `getLongRunningOperation(Method)` is not, even though it does not need any internal state? In general, having stateless helper class is much more flexible, than having a utility class with only-static methods or a singleton. For example mocking utility class with static methods is more difficult. 

In case if there is no very concrete reason to keep some methods static and some not, please convert `ProtoParser` to a stateless helper class (no static methods,  public constructor, no need to have `getProtoParser()`, no reason to be afraid of "overinstantiating" the class, everywhere when needed, it is done super fast and jvm is optimized for this kind of thing extremely well (unless you do it in ridiculous quantities, of course, but it is not the case)), this will allow keep things consistent and flexible (easy to mock/substitute with different implementation, if needed in certain cases).